maro/simulator/scenarios/vm_scheduling/business_engine.py

```diff
@@ -145,34 +171,157 @@ def _init_data(self):
             self._download_processed_data()
             logger.info_green("Data preparation is finished.")
 
-    def _cal_pm_amount(self) -> int:
-        amount: int = 0
-        for pm_type in self._config.PM:
-            amount += pm_type["amount"]
-
-        return amount
+    def _find_item(self, key: str, dictionary: dict) -> int:
+        for k, v in dictionary.items():
+            if k == key:
+                yield v
+            elif isinstance(v, list):
+                for item in v:
+                    for result in self._find_item(key, item):
+                        yield result
+            elif isinstance(v, dict):
+                for result in self._find_item(key, v):
+                    yield result
+
+    def _init_regions(self):
+        """Initialize the regions based on the config setting. The regions id starts from 0."""
+        self._regions = self._frame.regions
+        region_id = 0
+        zone_id = count(start=0)
+        for region_list in self._find_item("region", self._config.architecture):
+            for region_dict in region_list:
+                region = self._regions[region_id]
+                region.set_init_state(
+                    id=region_id
+                )
+                region.name = region_dict["name"]
+                region.zone_list = [next(zone_id) for _ in range(len(region_dict["zone"]))]
+                region.total_machine_num = sum(
+                    self._zones[zone_id].total_machine_num for zone_id in region.zone_list
+                )
+                region_id += 1
+
+    def _init_zones(self):
+        """Initialize the zones based on the config setting. The zone id starts from 0."""
+        self._zones = self._frame.zones
+        zone_id = 0
+        data_center_id = count(start=0)
+        for zone_list in self._find_item("zone", self._config.architecture):
+            for zone_dict in zone_list:
+                zone = self._zones[zone_id]
+                zone.set_init_state(
+                    id=zone_id
+                )
+                zone.name = zone_dict["name"]
```

Comment:
all assignment here that not saved in `xx.set_init_state()` should not be cleared in `reset()`, unless you re-call the `_init_xx` function in BE.reset()